### PR TITLE
varactor_junction_p_n_capacity

### PR DESCRIPTION
**Pull request by Zulusssss**
add law and tests

### CODE REVIEW

File: symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+from sympy import Eq, solve
+from symplyphysics import (units, Quantity, Symbol, validate_input, validate_output,
+                           dimensionless,)
+
+# Description
+## The principle of operation of the variator is based on the dependence of the barrier capacitance of the p-n junction on the voltage value.
+## Knowing the doping coefficient, the material parameter and the capacity without bias voltage, it is possible to calculate the capacity
+## of the varactor at a given voltage.
+
+## Law is: C =  C0 / (1 - V / V0)^y, where
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py:10]:
```suggestion
## Law is: C = C0 / (1 - V / V0)^y, where
```

File: symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+## C - barrier capacity of the p-n junction,
+## C0 - barrier capacitance of the p-n junction without bias voltage,
+## V - voltage,
+## V0 - material parameter,
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py:14]: What's this?

File: symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## y - doping coefficient.
+
+junction_capacitance = Symbol("junction_capacitance", units.capacitance)
+
+junction_transition_capacity_without_bias_voltage = Symbol("junction_transition_capacity_without_bias_voltage", units.capacitance)
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py:19]: Name it according to similar parameter: `junction_capacitance_without_bias_voltage`

File: symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+law = Eq(junction_capacitance, junction_transition_capacity_without_bias_voltage / (1 - voltage / material_parameter)**doping_coefficient)
+
+
+@validate_input(junction_transition_capacity_without_bias_voltage_=junction_transition_capacity_without_bias_voltage,
+                voltage_=voltage,
+                material_parameter_=material_parameter,
+                doping_coefficient_=doping_coefficient)
+@validate_output(junction_capacitance)
+def calculate_junction_capacitance(junction_transition_capacity_without_bias_voltage_: Quantity, voltage_: Quantity,
+                                                  material_parameter_: float, doping_coefficient_: Quantity) -> Quantity:
```
Comment (blackyblack) [inline symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py:33]: Coefficient is dimensionless and material parameter is quantity.

File: symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py (hunk 2 above)
Comment (Zulusssss) [reply symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py:14]: > What's this?

A certain constant having the dimension of the electric voltage and depending on the material.

Comment (blackyblack) [reply symplyphysics/laws/electricity/circuits/varactor_junction_p_n_capacity.py:14]: Add to description.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**